staticcheck/doc.go

```diff
@@ -332,6 +332,14 @@ the various parsing and formatting functions in \'strconv\'.`,
 		MergeIf:  lint.MergeIfAny,
 	},
 
+	"SA1031": {
+		Title:    `Overlapping byte slices passed to an encoder`,
+		Text:     `In an encoding function of the form Encode(dst, src), dst and src were found to reference the same memory`,
```

Comment:
We should probably explain _why_ that is an issue. It wouldn't be one if for every byte read we only wrote one byte.